odl/contrib/datasets/ct/mayo.py

```diff
@@ -233,25 +236,41 @@ def load_reconstruction(folder, slice_start=0, slice_end=-1):
         data_array = data_array.reshape([cols, rows], order='C')
         data_array = np.rot90(data_array, -1)
 
-        # Convert from CT numbers to densities
-        data_array /= 1024.0
+        # Convert from storage type to densities
+        # TODO: Optimize these computations
+        hu_values = (dataset.RescaleSlope * data_array +
+                     dataset.RescaleIntercept)
+        densities = (hu_values + 1000) / 1000
 
         # Store results
-        volumes.append(data_array)
+        volumes.append(densities)
         datasets.append(dataset)
 
     # Compute geometry parameters
     voxel_size = np.array(list(pixel_size) + [pixel_thickness])
     shape = np.array([rows, cols, len(volumes)])
     min_pt = (np.array(dataset.ImagePositionPatient) -
               np.array(dataset.DataCollectionCenterPatient))
-    min_pt[:2] += 0.5 * np.array(dataset.PixelSpacing)
-    max_pt = min_pt + voxel_size * np.array([rows - 1, cols - 1, 0])
 
+    max_pt = min_pt + voxel_size * np.array([rows, cols, 0])
+
+    # axis 1 has reversed convention
+    min_pt[1], max_pt[1] = -max_pt[1], -min_pt[1]
+
+    if len(datasets) > 1:
+        slice_distance = np.abs(
+            np.array(datasets[1].DataCollectionCenterPatient)[2] -
+            np.array(datasets[0].DataCollectionCenterPatient)[2])
+    else:
+        # If we only have one slice, we must approximate the distance.
+        slice_distance = pixel_thickness
+
+    # Set min and max points. We need half voxel offsets since dicom uses
+    # midpoint and ODL uses the corners.
     min_pt[2] = -np.array(datasets[0].DataCollectionCenterPatient)[2]
-    min_pt[2] -= 0.5 * pixel_thickness
+    min_pt[2] -= 0.5 * slice_distance
     max_pt[2] = -np.array(datasets[-1].DataCollectionCenterPatient)[2]
-    max_pt[2] += 0.5 * pixel_thickness
+    max_pt[2] += 0.5 * slice_distance
```

Comment:
What's the overall logic of computing the min/max in the z axis? Does it go from `-(patient z in dataset 0)` to 0, adjusted by half a slice distance in each direction?

Reply:
Okay, maybe a comment could make that more obvious, I had to think about it for a second. Maybe also the review markup was a bit to blame.